lib/Basic/FileSystem.cpp

```diff
@@ -30,6 +36,136 @@ namespace {
   };
 } // end anonymous namespace
 
+/// Does some simple checking to see if a temporary file can be written next to
+/// \p outputPath and then renamed into place.
+///
+/// Helper for swift::atomicallyWritingToFile.
+///
+/// If the result is an error, the write won't succeed at all, and the calling
+/// operation should bail out early.
```

Comment:
If the result is a bool instead of the error, does true mean good, or does true mean bad? It would help to say that here.